Rewrite docs with Doctocat

colebemis opened the pull request:
This pull request refactors the Primer CSS documentation site to use [Doctocat](http://github.com/primer/doctocat). The new site has feature parity with the old site (primer.style/css). 

TODO
- [x] Remove old docs
- [x] Update contributing docs
- [x] Make sure documentation is up to date
- [x] Test in Chrome
- [x] Test in Firefox
- [x] Test in Safari
- [x] Test in Edge

colebemis commented:
@broccolini and @emilybrick This is ready for your QA. The link to the latest deploy is https://primer-css-git-doctocat.primer.now.sh. 

The new site should have feature parity with https://primer.style/css _expect_ the new site doesn't have responsive search or [source code links](https://user-images.githubusercontent.com/4608155/62903985-6eaa5700-bd19-11e9-993a-a26bc0067295.png). Since responsive search is currently broken on https://primer.style/css, I don't think it's blocker for this pull.

colebemis commented:
@emilybrick I updated the example like we discussed 👍

colebemis commented:
> * the border x & y utility examples have no space between them, so it's hard to tell what each is doing. I'd suggest putting `mb-3` on them.

Since this is also a problem in primer.style/css, I vote we fix this in a future pull.

> * I think something is going on with the select menus on the [forms](https://primer-css-1wgiauo91.now.sh/css/components/forms) page 🤔

This should be fixed now ✅ The problem was that we weren't autoprefixing the styles in the live previews. I fixed that by using the `postcss-loader` to run `autoprefixer` on the styles before including them in the live preview iframes. 

 
> * On that note, I think they're missing a caret on the selects page as well:
>   [styleguide example](https://styleguide.github.com/primer/components/select-menu/#basic-example) vs [primer css example](https://primer-css-1wgiauo91.now.sh/css/components/select-menu#basic-example)

Since this is also a problem in primer.style/css, I vote we fix this in a future pull.

> * The color system docs link to the color utilities page, but link is [broken](https://github.com/primer/css/pull/851/files?file-filters%5B%5D=.md#diff-aa49489ab140fc7150ef21cadf1f8706R75) (I think there's an extra `/css/`

Fixed ✅

jonrohan commented on docs/gatsby-config.js:
```diff
@@ -0,0 +1,31 @@
+    {
+      resolve: 'gatsby-plugin-google-analytics',
+      options: {
+        trackingId: 'UA-126681523-2',
```
Gave you the wrong code

```suggestion
        trackingId: 'UA-126681523-1',
```

broccolini commented:
I want to make sure we have clarity on quality & shipping. I don't think we should deprecate the styleguide and point to primer.style/css until the new docs site is as good as or better than the old styleguide. I've been using both due to some features working better on one or the other. 

That doesn't mean fixing everything in this pr, just want to make sure you and @jonrohan coordinate!

Here's some quick feedback, can review more thoroughly tomorrow if needed, and not all of this is necessarily a blocker, just trying to capture issues as I find them. Reach out if you need help/clarification prioritizing:

- most animations are missing on https://primer-css-git-doctocat.primer.now.sh/css/utilities/animations (broken on all versions of the docs)
- double bullets in TOC showing on second levels, example https://primer-css-git-doctocat.primer.now.sh/css/support/typography and https://primer-css-git-doctocat.primer.now.sh/css/support/breakpoints
- black and white fade docs are missing from color system https://primer-css-git-doctocat.primer.now.sh/css/support/color-system, see here https://styleguide.github.com/primer/support/color-system/
- marketing variables should be at bottom of list under "support" since they extend base variables, it's odd to have them in between base variables. Also noticed that they're called marketing _variables_ not support here https://styleguide.github.com/primer/support/marketing-variables/ (honestly not sure what is more correct since it only includes spacing)
- "book" emoji not rendering here (broken on both sites) https://primer-css-git-doctocat.primer.now.sh/css/utilities/flexbox#required-reading
- no-wrap example needs updating so it flows over new wider width of docs https://primer-css-git-doctocat.primer.now.sh/css/utilities/flexbox#flex-wrap, see working example here: https://styleguide.github.com/primer/utilities/flexbox/
- marketing-layout docs completely different, unsure what it should be, please can you double check we didn't loose some content: https://primer-css-git-doctocat.primer.now.sh/css/utilities/marketing-layout vs https://styleguide.github.com/primer/utilities/marketing-layout/
- margin and padding examples lost colors to show the spacing, these were custom colors which can be replaced with utilities https://primer-css-git-doctocat.primer.now.sh/css/utilities/margin vs https://styleguide.github.com/primer/utilities/margin/
- breadcrumbs not showing, I think we still want them?
- status labels should link to status description
- I think we should replace "package" with "category" or something, it's been helpful to define core vs marketing etc. (we can discuss this further and add to future ship)

colebemis commented:
Thanks for this list, @broccolini! This is super helpful. I went through each item and noticed that all the issues you mentioned are also problems in `primer.style/css`. We should make GitHub issues for each item in your list and fix them before we deprecate the old styleguide but I don't think any of them are blockers for this PR. Let me know if that works for you.

That being said, I went ahead and fixed some of the easy issues you noticed:

> double bullets in TOC showing on second levels, example https://primer-css-git-doctocat.primer.now.sh/css/support/typography and https://primer-css-git-doctocat.primer.now.sh/css/support/breakpoints

This looks like a result of skipping from `h2` to `h4` in the markdown. I updated the headings in typography and breakpoints to remove the double bullets ✅ 

> marketing variables should be at bottom of list under "support" since they extend base variables, it's odd to have them in between base variables. Also noticed that they're called marketing variables not support here https://styleguide.github.com/primer/support/marketing-variables/ (honestly not sure what is more correct since it only includes spacing)

I moved "Marketing support" to the bottom of the Support section ✅

> no-wrap example needs updating so it flows over new wider width of docs https://primer-css-git-doctocat.primer.now.sh/css/utilities/flexbox#flex-wrap, see working example here: https://styleguide.github.com/primer/utilities/flexbox/

I set a width of `50%` to the flex container so the items always overflow ✅

colebemis commented:
I did some cross-browser testing and noticed a few issues:

## Edge

* Nav dropdowns are stuck open. We noticed the same issue on https://primer.style as well.

<img width="423" alt="Screen Shot 2019-08-14 at 3 37 04 PM" src="https://user-images.githubusercontent.com/4608155/63062189-d93fcc00-beac-11e9-950a-e267ac534bef.png">

* The SVGs on the homepage are _huge_.

<img width="954" alt="Screen Shot 2019-08-14 at 3 39 56 PM" src="https://user-images.githubusercontent.com/4608155/63062201-e6f55180-beac-11e9-83d7-846bc2d685d8.png">

## Safari

* Nav links in the nav drawer aren't clickable if they are below the fold.

![2019-08-14 at 4 04 PM](https://user-images.githubusercontent.com/4608155/63062327-510df680-bead-11e9-946a-0f2196dc76c8.gif)

---

I think all of these issues are important to fix before merging this PR.

colebemis commented:
> Nav dropdowns are stuck open. We noticed the same issue on https://primer.style as well.

Welp, looks like `<details>` isn't supported in Edge yet (https://caniuse.com/#feat=details) so we'll need to find a polyfill or find an alternative `<details>`.

colebemis commented on docs/content/components/alerts.md:
````diff
@@ -75,9 +78,12 @@ Add an icon to the left of the flash message to give it some funky fresh attenti
 Add a JavaScript enabled (via Crema) dismiss (close) icon on the right of any flash message.
 
-```erb
+```jsx live
````
```suggestion
```html live
```

colebemis commented on docs/content/components/box.md:
```diff
@@ -98,16 +95,17 @@ Rows can be used with or without `Box-header`, `Box-footer`, or `Box-body`.
     Box header
   </div>
-  <!-- This wrapping div ensures the first and last rows can be targeted for styling. -->
+  {/* This wrapping div ensures the first and last rows can be targeted for styling. */}
```
```suggestion
  <!-- This wrapping div ensures the first and last rows can be targeted for styling. -->
```

colebemis commented on docs/content/components/branch-name.md:
````diff
@@ -6,23 +6,22 @@ source: 'https://github.com/primer/css/tree/master/src/branch-name'
 You may also include an octicon before the branch name text:
 
-```erb title="Branch name with icon"
+```jsx live
````
```suggestion
```html live
```

colebemis commented on docs/content/components/buttons.md:
````diff
@@ -6,72 +6,79 @@ source: 'https://github.com/primer/css/tree/master/src/buttons'
-<button class="btn btn-danger" type="button">Danger button</button>
-<button class="btn btn-sm btn-danger" type="button">Small danger button</button>
+```html live
+<div>
````
```suggestion
```

colebemis commented on docs/content/components/buttons.md:
```diff
@@ -6,72 +6,79 @@ source: 'https://github.com/primer/css/tree/master/src/buttons'
+<div>
+  <button class="btn btn-danger" type="button">Danger button</button>
+  <button class="btn btn-sm btn-danger" type="button">Small danger button</button>
+</div>
```
```suggestion
```

colebemis commented on docs/content/components/buttons.md:
````diff
@@ -6,72 +6,79 @@ source: 'https://github.com/primer/css/tree/master/src/buttons'
-<button class="btn btn-primary" type="button">Primary button</button>
-<button class="btn btn-sm btn-primary" type="button">Small primary button</button>
+```html live
+<div>
````
```suggestion
```

colebemis commented on docs/content/components/buttons.md:
```diff
@@ -6,72 +6,79 @@ source: 'https://github.com/primer/css/tree/master/src/buttons'
+<div>
+  <button class="btn btn-primary" type="button">Primary button</button>
+  <button class="btn btn-sm btn-primary" type="button">Small primary button</button>
+</div>
```
```suggestion
```

colebemis commented on docs/content/components/buttons.md:
````diff
@@ -6,72 +6,79 @@ source: 'https://github.com/primer/css/tree/master/src/buttons'
-<button class="btn btn-outline" type="button">Outline button</button>
-<button class="btn btn-sm btn-outline" type="button">Outline button</button>
+```html live
+<div>
````
```suggestion
```

colebemis commented on docs/content/components/buttons.md:
```diff
@@ -6,72 +6,79 @@ source: 'https://github.com/primer/css/tree/master/src/buttons'
+<div>
+  <button class="btn btn-outline" type="button">Outline button</button>
+  <button class="btn btn-sm btn-outline" type="button">Outline button</button>
+</div>
```
```suggestion
```

colebemis commented on docs/content/components/buttons.md:
````diff
@@ -82,35 +89,45 @@ Use `.btn-large` with a type scale utility to transform the text to a bigger siz
-<button class="btn" type="button" disabled>Disabled button</button>
-<a class="btn disabled" href="#url" role="button">Disabled button</a>
+```html live
+<div>
````
```suggestion
```

colebemis commented on docs/content/components/buttons.md:
```diff
@@ -82,35 +89,45 @@ Use `.btn-large` with a type scale utility to transform the text to a bigger siz
+<div>
+  <button class="btn" type="button" disabled>Disabled button</button>
+  <a class="btn disabled" href="#url" role="button">Disabled button</a>
+</div>
```
```suggestion
```

colebemis commented on docs/content/components/buttons.md:
```diff
@@ -82,35 +89,45 @@ Use `.btn-large` with a type scale utility to transform the text to a bigger siz
+<div>
+  <button class="btn btn-primary" type="button" disabled>Disabled button</button>
+  <a class="btn btn-primary disabled" href="#url" role="button">Disabled button</a>
+</div>
```
```suggestion
```

colebemis commented on docs/content/components/buttons.md:
````diff
@@ -82,35 +89,45 @@ Use `.btn-large` with a type scale utility to transform the text to a bigger siz
-<button class="btn btn-primary" type="button" disabled>Disabled button</button>
-<a class="btn btn-primary disabled" href="#url" role="button">Disabled button</a>
+```html live
+<div>
````
```suggestion
```

colebemis commented on docs/content/components/buttons.md:
````diff
@@ -82,35 +89,45 @@ Use `.btn-large` with a type scale utility to transform the text to a bigger siz
-<button class="btn btn-danger" type="button" disabled>Disabled button</button>
-<a class="btn btn-danger disabled" href="#url" role="button">Disabled button</a>
+```html live
+<div>
````
```suggestion
```

colebemis commented on docs/content/components/buttons.md:
```diff
@@ -82,35 +89,45 @@ Use `.btn-large` with a type scale utility to transform the text to a bigger siz
+<div>
+  <button class="btn btn-danger" type="button" disabled>Disabled button</button>
+  <a class="btn btn-danger disabled" href="#url" role="button">Disabled button</a>
+</div>
```
```suggestion
```

colebemis commented on docs/content/components/buttons.md:
````diff
@@ -82,35 +89,45 @@ Use `.btn-large` with a type scale utility to transform the text to a bigger siz
-<button class="btn btn-outline" type="button" disabled>Disabled button</button>
-<a class="btn btn-outline disabled" href="#url" role="button">Disabled button</a>
+```html live
+<div>
````
```suggestion
```

colebemis commented on docs/content/components/buttons.md:
```diff
@@ -82,35 +89,45 @@ Use `.btn-large` with a type scale utility to transform the text to a bigger siz
+<div>
+  <button class="btn btn-outline" type="button" disabled>Disabled button</button>
+  <a class="btn btn-outline disabled" href="#url" role="button">Disabled button</a>
+</div>
```
```suggestion
```

colebemis commented on docs/content/components/buttons.md:
````diff
@@ -82,35 +89,45 @@ Use `.btn-large` with a type scale utility to transform the text to a bigger siz
 ```html
-<p><button class="btn btn-block" type="button">Block button</button></p>
-<p><button class="btn btn-sm btn-block" type="button">Small block button</button></p>
+<div>
````
```suggestion
```

colebemis commented on docs/content/components/buttons.md:
```diff
@@ -82,35 +89,45 @@ Use `.btn-large` with a type scale utility to transform the text to a bigger siz
+<div>
+  <p><button class="btn btn-block" type="button">Block button</button></p>
+  <p><button class="btn btn-sm btn-block" type="button">Small block button</button></p>
+</div>
```
```suggestion
```

colebemis commented on docs/content/components/buttons.md:
````diff
@@ -141,55 +158,59 @@ You can easily append a count to a **small button**. Add the `.with-count` class
-  <span class="Counter">12</span>
-</button>
+```html live
+<div>
````
```suggestion
```

colebemis commented on docs/content/components/buttons.md:
```diff
@@ -141,55 +158,59 @@ You can easily append a count to a **small button**. Add the `.with-count` class
+    Button
+    <span class="Counter">12</span>
+  </button>
+</div>
```
```suggestion
```

colebemis commented on docs/content/components/buttons.md:
````diff
@@ -141,55 +158,59 @@ You can easily append a count to a **small button**. Add the `.with-count` class
-  <button class="btn BtnGroup-item btn-sm" type="button">Button</button>
-  <button class="btn BtnGroup-item btn-sm" type="button">Button</button>
+```html live
+<div>
````
```suggestion
```

colebemis commented on docs/content/components/buttons.md:
```diff
@@ -141,55 +158,59 @@ You can easily append a count to a **small button**. Add the `.with-count` class
+    <button class="btn BtnGroup-item btn-sm" type="button">Button</button>
+    <button class="btn BtnGroup-item btn-sm" type="button">Button</button>
+    <button class="btn BtnGroup-item btn-sm" type="button">Button</button>
+  </div>
```
```suggestion
```

colebemis commented on docs/content/components/forms.md:
````diff
@@ -145,35 +148,37 @@ Primer CSS adds light `height` and `vertical-align` styles to `<select>`s for al
-  <option>Battlestar Galactica</option>
-</select>
+```html live
+<div>
````
```suggestion
```

colebemis commented on docs/content/components/forms.md:
```diff
@@ -145,35 +148,37 @@ Primer CSS adds light `height` and `vertical-align` styles to `<select>`s for al
+    <option>Bears</option>
+    <option>Battlestar Galactica</option>
+  </select>
+</div>
```
```suggestion
```

colebemis commented on docs/postcss.config.js:
```diff
@@ -0,0 +1,10 @@
+module.exports = {
```
- [ ] Add a comment explaining why this file is necessary

colebemis commented on docs/src/@primer/gatsby-theme-doctocat/live-code-scope.js:
```diff
@@ -0,0 +1,3 @@
+import Octicon, {getIconByName} from '@primer/octicons-react'
+import {StyledOcticon} from '@primer/components'
+export default {Octicon, StyledOcticon, getIconByName}
```
These probably don't need to be in scope anymore

colebemis commented on docs/content/components/alerts.md:
````diff
@@ -64,9 +66,10 @@ Add `.flash-warn`, `.flash-error`, or `.flash-success` to the flash message to m
 Add an icon to the left of the flash message to give it some funky fresh attention.
 
-```erb
+```jsx live
````
```suggestion
```html live
```

colebemis commented:
> Safari: Nav links in the nav drawer aren't clickable if they are below the fold.

I spent most of yesterday trying to fix this bug. Turns out there is no straightforward fix. After discussing with @emplums, I think it's okay if we move forward with this pull without fixing this since it only affects macOS Safari users with a viewport of <1000px.

I filed an issue for this bug on the primer/dococat repo (https://github.com/primer/doctocat/issues/56) and react-focus-lock repo (https://github.com/theKashey/react-focus-lock/issues/79).
